docs/configuration.rst

```diff
@@ -42,9 +42,129 @@ File: config.yml
 manage :term:`Services` components.
-A basic `config.example.yml`_ file is provided, for sample definition of expected schemas per service. Please refer to
-its comment header for specific format and parameter
-details.
+A basic `config.example.yml`_ file is provided, for sample definition of expected schemas per service.
+
+This config file contains the following required sections :
+
+services:
+#########
+
+The ``services`` section contains the definition of managed services by Cowbird.
+Each service is provided as a string that must match an actual implementation in Cowbird.
+Each service must be further configured with one or more of the following parameters. If
+a required parameter is missing for a service it will throw a ``ServiceConfigurationException`` exception.
+
+.. code-block:: yaml
+
+  service:
+    active:             [optional, default=False] bool allowing to deactivate a service and stop managing it
+    priority:           [optional, default=math.inf] Relative priority between services while handling events
+                        Lower values have higher priority, default value is last
+    url:                [optional, default=None] URI of the web service represented by this Cowbird service.
+                        Some Cowbird services do not represent web service, but other will throw an exception if
+                        missing.
+    workspace_dir:      [optional, default=None] Location of the users workspace root.
+                        Required for the following services : FileSystem, Catalog and Geoserver
+
+sync_permissions:
+#################
+
+This section defines how to synchronize permissions between services when they share resources.
+This is used only for the synchronization of permissions between Magpie services/resources.
+The ``sync_permissions`` are defined first by a list of Magpie services and their associated resources,
+defined in the ``services`` section below. The mappings defining how the resources should be synchronized are described
+in the other section ``permissions_mapping``.
+
+services:
+_________
+
+This section contains the different resources that can be synchronized, ordered by service. The services found in
+this section of the config should also exist in Magpie.
+
+For more details on available services on Magpie, refer to these pages :
+
+- `Magpie available services <https://pavics-magpie.readthedocs.io/en/latest/services.html#available-services>`_
+- `Magpie services API <https://pavics-magpie.readthedocs.io/en/latest/autoapi/magpie/services/index.html>`_
+
+Each service defines a list of resource keys, which are custom names that define a resource path.
+They should correspond to the names used in the ``permissions_mapping`` section below.
+Each resource path contains the list of its segments, with their corresponding name and type.
+
+The name of a segment can either be a string name, a variable or a ``MULTI_TOKEN`` (``**``).
```

Comment:
You can use a reference: 
```rst
:data:`cowbird.[...].MULTI_TOKEN` 
```
It will point directly to the actual definition in the code.
You can also add a `.. py:currentmodule:: cowbird.[...]` before the `:data:` reference to link only with `MULTI_TOKEN` word without the full python object path.